app/models/agents/read_file_agent.rb

```diff
@@ -0,0 +1,49 @@
+module Agents
+  class ReadFileAgent < Agent
+    include FormConfigurable
+    include FileHandling
+
+    cannot_be_scheduled!
+
+    def default_options
+      {
+        'data_key' => 'data'
+      }
+    end
+
+    description do
+      <<-MD
+        The ReadFileAgent takes events from `FileHandling` agents, reads the file, and emits the contents as a string.
```

Comment:
We should be able to access the file size for the three sources that are supported at the moment. The limit we currently have is the same as for other agents: `DELAYED_JOB_MAX_RUNTIME * available_bandwidth`.